### ISSUE TITLE
Streampager abort when piping files into xargs hg blame

### ISSUE THREAD

**Issue by akushner** [markbt/streampager#36]
Hello Mark!

While trying to do some blames via xargs, got an abort. See below.

I was using EdenSCM 4.4.2_20210223_123033_737026f39597_1.fb.el_737026f3.

Jun said that he's seen this type of abort, also.

Failure was repeatable. 

```
$ cd fbcode/eden/edenscm && find . -type f -name "*.py"| xargs hg blame
  simpkins 2019-11-01: # Copyright (c) Facebook, Inc. and its affiliates.
thread '<unnamed>' panicked at 'range end index 323 out of range for slice of length 283', third-party/rust/vendor/streampager-0.8.0/src/file.rs:493:26
                        note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
                                                                                                     xargs: hg: exited with status 255; aborting
```

(Worked around the crash using --pager=off)

**Comment by markbt** [comment]
I've also seen this before but didn't have a reliable repro.  The change in b562cb0440 should be the fix.  I'll upgrade EdenSCM to the latest streampager version to see if that works.

**Comment by markbt** [comment]
Closing, as this should be fixed by b562cb044029a9f67512d4a5ca8bf6fd3a490f30.
